java/ql/src/semmle/code/java/frameworks/ratpack/RatpackExec.qll

```diff
@@ -0,0 +1,160 @@
+import java
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.FlowSteps
+
+/** A reference type that extends a parameterization the Promise type. */
+private class RatpackPromise extends RefType {
+  RatpackPromise() {
+    getSourceDeclaration().getASourceSupertype*().hasQualifiedName("ratpack.exec", "Promise")
+  }
+}
+
+private class RatpackPromiseValueMethod extends Method, TaintPreservingCallable {
+  RatpackPromiseValueMethod() { isStatic() and hasName("value") }
```

Comment:
This is not restricting the declaring type - i.e. nothing ties this to Ratpack, it's just matching methods named "value".